status/status_history.go

```diff
@@ -8,19 +8,26 @@ import (
 	"time"
 
 	"github.com/juju/errors"
+	"github.com/juju/utils/set"
 )
 
 // StatusHistoryFilter holds arguments that can be use to filter a status history backlog.
 type StatusHistoryFilter struct {
-	Size  int
-	Date  *time.Time
+	// Size indicates how many results are expected at most.
+	Size int
+	// FromDate indicates the earliest date from which logs are expected.
+	FromDate *time.Time
+	// Delta indicates the age of the oldest log expected.
 	Delta *time.Duration
+	// Exclude indicates the status messages that should be excluded
+	// from the returned result.
+	Exclude set.Strings
 }
 
 // Validate checks that the minimum requirements of a StatusHistoryFilter are met.
 func (f *StatusHistoryFilter) Validate() error {
 	s := f.Size > 0
-	t := f.Date != nil
+	t := f.FromDate != nil
```

Comment:
Noticed validation code just below here:
return errors.NotValidf("empty struct")

"empty struct" means nothing to an end user.
"missing filter parameters" or something like that is more like what we'd want